### PR TITLE
[Merged by Bors] - chore: Remove `Init.Propext`

### PR DESCRIPTION
**Pull request by YaelDillies**
These lemmas can easily go to `Logic.Basic`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — APPROVED** [review]
Looks plausible to me.

File: Mathlib/Logic/Basic.lean [hunk 1]
```diff
@@ -923,6 +923,26 @@ theorem forall_prop_congr' {p p' : Prop} {q q' : p → Prop} (hq : ∀ h, q h 
...
+  propext (imp_congr h₁.to_iff h₂.to_iff)
+
+lemma imp_congr_ctx_eq {a b c d : Prop} (h₁ : a = c) (h₂ : c → b = d) : (a → b) = (c → d) :=
+  propext (imp_congr_ctx h₁.to_iff fun hc ↦ (h₂ hc).to_iff)
+
+lemma eq_true_intro (h : a) : a = True := propext (iff_true_intro h)
+lemma eq_false_intro (h : ¬a) : a = False := propext (iff_false_intro h)
+
+/-- Alias of `propext`. -/
+lemma Iff.eq : (a ↔ b) → a = b := propext
```
Comment (digama0) [inline Mathlib/Logic/Basic.lean:938]:
```suggestion
alias Iff.eq := propext
```

Comment (YaelDillies) [reply Mathlib/Logic/Basic.lean:938]: I can't because `alias` is not imported there. Should I import it?

Comment (digama0) [reply Mathlib/Logic/Basic.lean:938]: yes

Comment (YaelDillies) [reply Mathlib/Logic/Basic.lean]: Nevermind, it is? 🤔

**Review by jcommelin — COMMENTED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

File: Mathlib/Logic/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Logic/Basic.lean]: Sorry I remember the problem: `alias` creates `def Iff.eq := propext`, not `lemma Iff.eq := propext`. Will nolint for now.

Comment (digama0) [reply Mathlib/Logic/Basic.lean]: it shouldn't? Report it as a bug if so

Comment (digama0) [reply Mathlib/Logic/Basic.lean]: It's supposed to create defs from defs and theorems from theorems, but you are aliasing an `axiom` and I guess these can be either defs or theorems

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**